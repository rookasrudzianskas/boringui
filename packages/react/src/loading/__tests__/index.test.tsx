import React from "react";
import {mount} from "enzyme";

import Loading from "../index";

describe("Loading", () => {
  it("should render correctly", () => {
    const wrapper = mount(<Loading />);

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it("should render with default aria-label", () => {
    const wrapper = mount(<Loading />);

    expect(wrapper.find(".nextui-loading").at(0).prop("aria-label")).toEqual("Loading");
  });

  it("should render with default aria-label for spinner", () => {
    const wrapper = mount(<Loading type="spinner" />);

    expect(wrapper.find(".nextui-spinner-container").at(0).prop("aria-label")).toEqual("Loading");
  });

  it("should work with children", () => {
    const wrapper = mount(<Loading>Loading</Loading>);

    expect(wrapper.find(".nextui-loading-label").at(0).text()).toContain("Loading");
  });

  it("should work with different colors", () => {
    const wrapper = mount(
      <div>
        <Loading color="primary" />
        <Loading color="success" />
        <Loading color="secondary" />
        <Loading color="warning" />
        <Loading color="error" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it("should work with different text colors", () => {
    const wrapper = mount(
      <div>
        <Loading textColor="primary" />
        <Loading textColor="success" />
        <Loading textColor="secondary" />
        <Loading textColor="warning" />
        <Loading textColor="error" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it("should work with different sizes", () => {
    const wrapper = mount(
      <div>
        <Loading size="xs" />
        <Loading size="sm" />
        <Loading size="md" />
        <Loading size="lg" />
        <Loading size="xl" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it("should work with different types", () => {
    const wrapper = mount(
      <div>
        <Loading type="default" />
        <Loading type="spinner" />
        <Loading type="points" />
        <Loading type="points-opacity" />
        <Loading type="gradient" />
      </div>,
    );

    expect(wrapper.html()).toMatchSnapshot();
    expect(() => wrapper.unmount()).toMatchSnapshot();
  });

  it("should work with text in spinner type", () => {
    const wrapper = mount(<Loading type="spinner">Spinner</Loading>);

    expect(wrapper.find(".nextui-spinner").at(0).text()).toContain("Spinner");
  });
});
