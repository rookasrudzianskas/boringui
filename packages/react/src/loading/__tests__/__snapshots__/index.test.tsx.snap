// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `"<div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div>"`;

exports[`Loading should render correctly 2`] = `[Function]`;

exports[`Loading should work with different colors 1`] = `"<div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-primary nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-bdkpVZ-color-success nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-erOfwU-color-secondary nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-eVEedQ-color-warning nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-dGjLsC-color-error nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different colors 2`] = `[Function]`;

exports[`Loading should work with different sizes 1`] = `"<div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-bboDzR-size-xs nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-eWttbs-size-sm nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-eWTJwO-size-lg nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-fYAaIu-size-xl nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different sizes 2`] = `[Function]`;

exports[`Loading should work with different text colors 1`] = `"<div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-jjYXKD-textColor-primary\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-lgCEhg-textColor-success\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-iQizOB-textColor-secondary\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fgUECR-textColor-warning\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-jYiaFD-textColor-error\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different text colors 2`] = `[Function]`;

exports[`Loading should work with different types 1`] = `"<div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-jJwoeB-type-default nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-default\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><div class=\\"nextui-c-fpphdg nextui-c-fpphdg-fPWnWX-size-md nextui-spinner\\"><div aria-label=\\"Loading\\" class=\\"nextui-c-jTKaZN nextui-spinner-container\\"><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span><span class=\\"nextui-c-jdsqRm\\"></span></div></div></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-cGhcKQ-type-points nextui-c-chTLhx-dhbJvz-cv nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-points\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-heRREU-type-points-opacity nextui-c-chTLhx-dhbJvz-cv nextui-c-chTLhx-iPJLV-css nextui-loading nextui-loading-points-opacity\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div><div class=\\"nextui-c-bOoPrE nextui-c-bOoPrE-iWdsRK-color-default nextui-c-bOoPrE-fQoxSm-textColor-default\\"><span aria-label=\\"Loading\\" class=\\"nextui-c-chTLhx nextui-c-chTLhx-cPGQpm-size-md nextui-c-chTLhx-mBJPy-type-gradient nextui-c-chTLhx-icvvata-css nextui-loading nextui-loading-gradient\\"><i class=\\"_1\\"></i><i class=\\"_2\\"></i><i class=\\"_3\\"></i></span></div></div>"`;

exports[`Loading should work with different types 2`] = `[Function]`;
